sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SQLHadoopMapReduceCommitProtocol.scala

```diff
@@ -66,4 +68,18 @@ class SQLHadoopMapReduceCommitProtocol(
     logInfo(s"Using output committer class ${committer.getClass.getCanonicalName}")
     committer
   }
+
+  override def newTaskTempFile(
```

Comment:
Please help me understand how to ensure that the check that you have added is applicable for all forms of commit Protocol that could be added in the future. For example, PathOutputCommitProtocol[https://github.com/hortonworks-spark/cloud-integration/blob/master/spark-cloud-integration/src/main/site/markdown/index.md](url) might not enter this flow at all.

Reply:
The limitation only configures in SQLConf. So it only ensures the `SQLHadoopMapReduceCommitProtocol`. It cannot be used in any non-sql package implementation. I will describe it in the configuration doc.

Reply:
Do we have any limitation like this before? I can help me recollect case where the config is only for SQL and not for other modes of operation?